From 8c585df6b8550ba5321f0ce872ada45b5ab80610 Mon Sep 17 00:00:00 2001
From: Li Guohui <liguohui2061@phytium.com.cn>
Date: Thu, 16 May 2024 15:42:54 +0800
Subject: [PATCH 046/360] dt-bindings: media: Add binding for Phytium JPEG
 Encoder Engine

Add device tree binding documentation for the JPEG Encoder Engine of Phytium SoCs.

Signed-off-by: Li Guohui <liguohui2061@phytium.com.cn>
Signed-off-by: Wang Min <wangmin@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
---
 .../bindings/media/phytium,jpeg.yaml          | 52 +++++++++++++++++++
 1 file changed, 52 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/phytium,jpeg.yaml

diff --git a/Documentation/devicetree/bindings/media/phytium,jpeg.yaml b/Documentation/devicetree/bindings/media/phytium,jpeg.yaml
new file mode 100644
index 000000000000..71fd143130c9
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/phytium,jpeg.yaml
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/phytium,jpeg.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium JPEG Engine
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+description: |-
+  The JPEG Engine embedded in the Phytium SOCs can capture
+  and compress video data from digital or analog sources.
+
+properties:
+  compatible:
+    const: phytium,jpeg
+
+  reg:
+    maxItems: 1
+    description: |
+      Contains the offset and length of the JPEG Engine memory region.
+
+  interrupts:
+    maxItems: 1
+    description: |
+      The interrupt associated with the VE on this platform.
+
+  phytium,ocm-buf-addr:
+    $ref: /schemas/types.yaml#/definitions/uint64
+    description: |
+      The physical address used to storage the inputing video data.
+
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - phytium,ocm-buf-addr
+
+additionalProperties: false
+
+examples:
+  - |
+    jpeg: jpeg@32b32000 {
+      compatible = "phytium,jpeg";
+      reg = <0x0 0x32b32000 0 0x1000>;
+      interrupts = <0 41 4>;
+      phytium,ocm-buf-addr = <0x30c40000 0x30c60000>;
+      status = "okay";
+    };
-- 
2.20.1

